hibernate-reactive-core/src/test/java/org/hibernate/reactive/timezones/TimeZoneStorageMappingTests.java

```diff
@@ -0,0 +1,308 @@
+						  assertThat( resultList.get( 0 ).get( 11, Duration.class ) ).isEqualTo( Duration.ofHours( 1L ) );
+					  } )
+				)
+		);
+	}
+
+	public void testOffsetRetainedFormat(VertxTestContext context, String suffix) {
+		test( context, openSession()
+				.thenCompose( session -> session.createQuery(
+							  "select " +
+									  "format(e.offsetTime" + suffix + " as 'HH:mm:ssxxx'), " +
+									  "format(e.offsetDateTime" + suffix + " as 'dd/MM/yyyy ''at'' HH:mm:ssxxx'), " +
+									  "format(e.zonedDateTime" + suffix + " as 'dd/MM/yyyy ''at'' HH:mm:ssxxx'), " +
+									  "1 from TimeZoneStorageEntity e " +
+									  "where e.offsetDateTime" + suffix + " = e.offsetDateTime" + suffix,
+							  Tuple.class
+					  ).getResultList()
+					  .thenAccept( resultList -> {
+						  assertThat( resultList.get( 0 ).get( 0, String.class ) ).isEqualTo( TIME_FORMATTER.format( OFFSET_TIME ) );
+						  assertThat( resultList.get( 0 ).get( 1, String.class ) ).isEqualTo( FORMATTER.format( OFFSET_DATE_TIME ) );
+						  assertThat( resultList.get( 0 ).get( 2, String.class ) ).isEqualTo( FORMATTER.format( ZONED_DATE_TIME ) );
+					  } )
+				)
+		);
+	}
+
+	@Test
+	public void testNormalize(VertxTestContext context) {
+		test( context, openSession()
+				.thenCompose( session -> session.createQuery(
+							"select " +
+									"e.offsetTimeNormalized, " +
+									"e.offsetDateTimeNormalized, " +
+									"e.zonedDateTimeNormalized, " +
+									"e.offsetTimeNormalizedUtc, " +
+									"e.offsetDateTimeNormalizedUtc, " +
+									"e.zonedDateTimeNormalizedUtc " +
+									"from TimeZoneStorageEntity e",
+							Tuple.class
+					).getResultList()
+					.thenAccept( resultList -> {
+						assertThat( resultList.get( 0 ).get( 0, OffsetTime.class ).toLocalTime()).isEqualTo( OFFSET_TIME.withOffsetSameInstant( JVM_TIMEZONE_OFFSET ).toLocalTime() );
+						assertThat( resultList.get( 0 ).get( 0, OffsetTime.class ).getOffset()).isEqualTo( JVM_TIMEZONE_OFFSET );
+						assertThat( resultList.get( 0 ).get( 1, OffsetDateTime.class ).toInstant()).isEqualTo( OFFSET_DATE_TIME.toInstant() );
+						assertThat( resultList.get( 0 ).get( 2, ZonedDateTime.class ).toInstant()).isEqualTo( ZONED_DATE_TIME.toInstant() );
+						assertThat( resultList.get( 0 ).get( 3, OffsetTime.class ).toLocalTime()).isEqualTo( OFFSET_TIME.withOffsetSameInstant( ZoneOffset.UTC ).toLocalTime() );
+						assertThat( resultList.get( 0 ).get( 3, OffsetTime.class ).getOffset()).isEqualTo( ZoneOffset.UTC );
+						assertThat( resultList.get( 0 ).get( 4, OffsetDateTime.class ).toInstant()).isEqualTo( OFFSET_DATE_TIME.toInstant() );
+						assertThat( resultList.get( 0 ).get( 5, ZonedDateTime.class ).toInstant()).isEqualTo( ZONED_DATE_TIME.toInstant() );
+					}
+				)
+			)
+		);
+	}
+
+	// This converted test fails during the getResultList()
+	// Unsure how to enable the "@RequiresDialectFeature(....)" annotation which is an ORM JUnit construct
```

Comment:
How does the annotation check if the dialect is valid or not?

Reply:
[DialectFeatureChecks](https://github.com/hibernate/hibernate-orm/blob/main/hibernate-testing/src/main/java/org/hibernate/testing/orm/junit/DialectFeatureChecks.java) but it's over in their `hibernate-testing` package

This class is coded with options/features for all the dialects.  The @Disable'd test can't access their testing src.

Reply:
JUnit 5 has the annotation `@DisableIf`. If you check the condition, we can figure out to add it to our test.
I don't think we need something as sophisticated as the one in Hibernate ORM (for now at least).

Reply:
We should be able to get the dialect from the factory

Reply:
The existing `@DisabledIf` annotation use text-based scripting language...  accessing system properties.

I did find an [example](https://javabydeveloper.com/junit-5-conditional-test-execution-with-examples/) of writing custom `ExecutionCondition` classes that can use java code, which currently resides in specific Dialects and `DialectFeatureChecks`.  Seems like a lot of work right now.

Reply:
.. check that

I created a `doNotRunWithDialect()` method in the test using ORM's logic for their checks.

`@DisabledIf( "doNotRunWithDialect" )`

Reply:
I will have to look into it, but please, try to use better naming.
It's not clear at all what `doNotRunWithDialect` means.


In this case, this might make more sense:
```
@EnableIf("supportsTimeZoneStorage")
```

Note that I'm suggesting some ideas, I haven't look into this issue in depth.
We could also create an extension if it's easier.